game-core/src/main/java/games/strategy/triplea/ai/pro/ProNonCombatMoveAi.java

```diff
@@ -526,6 +525,12 @@ private void determineIfMoveTerritoriesCanBeHeld() {
         ProMatches.territoryHasFactoryAndIsNotConqueredOwnedLand(player, data));
     seaFactories = CollectionUtils.getMatches(seaFactories,
         ProMatches.territoryHasInfraFactoryAndIsOwnedLandAdjacentToSea(player, data));
+    final Set<Territory> territoriesToCheck = new HashSet<>(seaFactories);
+    for (final Territory t : seaFactories) {
```

Comment:
This looks to be another parallel stream opportunity. I'm not sure how much it helps performance here, but FWIW here is the parallel stream equivalent:
```
    territoriesToCheck.addAll(
        seaFactories.parallelStream()
            .map(t -> data.getMap().getNeighbors(
                t, ProMatches.territoryCanMoveSeaUnits(player, data, true)))
            .flatMap(Collection::stream)
            .collect(Collectors.toSet()));
```

Reply:
Doesn't really make a difference here as this method is super fast and just finding which territories need checked for TerritoryValueUtils.